src/wui/interactive_player.cc

```diff
@@ -101,6 +101,7 @@ void draw_bobs_for_visible_field(const Widelands::EditorGameBase& egbase,
                                  const InfoToDraw info_to_draw,
                                  const Widelands::Player& player,
                                  RenderTarget* dst) {
+	MutexLock m(MutexLock::ID::kObjects);
 	for (Widelands::Bob* bob = field.fcoords.field->get_first_bob(); bob;
```

Comment:
How many bobs are on an average map? Locking a long loop might not be the best option. 
If the drawing is the problem, maybe a MutexLock per iteration would be better.